packages/protocol/contracts/stability/PairManager.sol

```diff
@@ -0,0 +1,290 @@
+pragma solidity ^0.5.13;
+pragma experimental ABIEncoderV2;
+
+import { IPairManager } from "./interfaces/IPairManager.sol";
+import { IReserve } from "./interfaces/IReserve.sol";
+import { IMentoExchange } from "./interfaces/IMentoExchange.sol";
+
+import { StableToken } from "./StableToken.sol";
+
+import { UsingRegistry } from "../common/UsingRegistry.sol";
+import { Initializable } from "../common/Initializable.sol";
+import { FixidityLib } from "../common/FixidityLib.sol";
+
+// TODO: Remove when migrating to mento-core. Newer versions of OZ-contracts have this interface
+interface IERC20Metadata {
+  function symbol() external view returns (string memory);
+}
+
+/**
+ * @title PairManager
+ * @notice The pair manager allows the creation of virtual pairs that are tradeable via Mento.
+ */
+contract PairManager is IPairManager, Initializable, UsingRegistry {
+  using FixidityLib for FixidityLib.Fraction;
+
+  /* ==================== State Variables ==================== */
+
+  // Address of the broker contract.
+  address public broker;
+
+  // Maps a pair id to the corresponding pair struct.
+  // pairId is in the format "stableSymbol:collateralSymbol:exchangeName"
+  mapping(bytes32 => Pair) public pairs;
+  // Tracks whether or not a pair with the given pairId exists.
+  mapping(bytes32 => bool) public isPair;
```

Comment:
We could remove this by checking if `pairs[pairId].stableAsset != address(0)` and save one storage read on `getPair`